### PR TITLE
Added Pagination to rafty

### PR DESCRIPTION
**Pull request by Gajendra-singh-r**
- [x] Created pagination component
- [x] Added `docs` for pagination 
- [x] Created `storybook` for pagination

### CODE REVIEW

File: apps/docs/app/docs/components/pagination/page.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { Pagination } from "@rafty/ui";
+
+<Pagination />;
+```
+
+## Usage
+
+{% example name="pagination:usage" /%}
+
+## size
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/pagination/page.md:33]: Write it in capitalized casing

File: apps/docs/app/docs/components/pagination/page.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+{% example name="pagination:size" /%}
+
+## showQuickJumper
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/pagination/page.md:37]: Write it in capitalized casing

File: apps/docs/app/docs/components/pagination/page.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+{% example name="pagination:showQuickJumper" /%}
+
+## total_number
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/pagination/page.md:41]: Write it in capitalized casing

File: apps/docs/app/docs/components/pagination/page.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+{% example name="pagination:total_number" /%}
+
+## showSizeChanger
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/pagination/page.md:45]: Write it in capitalized casing

File: apps/docs/app/docs/components/pagination/page.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+| defaultCurrent   | Default current page number.                                     | `number`                                             | 1          |
+| defaultPageSize  | Default page size.                                               | `number`                                             | 10         |
+| disabled         | Whether the pagination is disabled.                              | `boolean`                                            |            |
+| hideOnSinglePage | Whether to hide pagination on a single page.                     | `boolean`                                            | false      |
+| showQuickJumper  | Whether to show a quick jumper input or customize the go button. | `boolean`                                            | false      |
+| showSizeChanger  | Whether to show the page size changer.                           | `boolean`                                            |            |
+| showTotal        | Callback function to display total and range info.               | `(total:number,range:string) => ReactNode or string` |            |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/pagination/page.md:72]: Pass dash ( - ) in default column if there is no default

File: apps/docs/lib/navigation.ts [hunk 6]
```diff
@@ -99,6 +99,10 @@ export const navigation = [
         title: "Popover",
         href: "/docs/components/popover",
       },
+      {
+        title: "Pagination",
+        href: "/docs/components/pagination",
+      },
```
Comment (LuckySharma014) [inline apps/docs/lib/navigation.ts:105]: Position this before password field

**Review by LuckySharma014 — CHANGES_REQUESTED** [review]
Did a overview review for now but will do a brief review after they are done with requested changes

File: libs/ui/src/pagination/pagination.tsx
[the remark this answers is not in the document]
Comment (LuckySharma014) [reply libs/ui/src/pagination/pagination.tsx:1]: You need to change the file casing via github.com

File: libs/ui/src/pagination/pagination.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    const onChangeHandle = (page: number, size: number) => {
+      setCurrentPage(page);
+      setPageSize(size);
+      onChange?.(page, size);
+    };
+
+    const onPageSizeChange = (value: number) => onChangeHandle(1, value);
+    const onPrev = () => onChangeHandle(currentPage - 1, page);
+    const onNext = () => onChangeHandle(currentPage + 1, page);
+    const onPageChange = (value: number) => onChangeHandle(value, page);
```
[the remark this answers is not in the document]
Comment (LuckySharma014) [reply libs/ui/src/pagination/pagination.tsx]: Still variable names are very confusing

File: libs/ui/src/pagination/PaginationButtons.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        size="icon"
+        onClick={onPrev}
+        isDisabled={prev_disabled}
+      >
+        <ChevronLeftIcon
+          className={classNames(
+            size === "sm" && "h-3.5 w-3.5",
+            size === "md" && "h-5 w-5",
+            size === "lg" && "h-6 w-6",
+          )}
```
[the remark this answers is not in the document]
Comment (LuckySharma014) [reply libs/ui/src/pagination/PaginationButtons.tsx]: You have made class variable as internal variable which is not the convention we use

**Review by MathurAditya724 — APPROVED** [review]

**Review by LuckySharma014 — APPROVED** [review]

**Pull request merged**